engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java

```diff
@@ -1601,4 +1388,18 @@ private Object getBeanPropertyValue(ValidationContext<?> validationContext, Obje
 
 		return value;
 	}
+
+	private String getPropertyName(ConstraintLocation location) {
```

Comment:
I like https://github.com/hibernate/hibernate-validator/pull/577/commits/4fe2bfb21d7c40054371c99446cf096bd7aedfe2 but it's a bit unfortunate we have to duplicate this logic several times.